File: server/omi/socket/socket-client.go
/**
* 本文件由Omi.js自动生成，谨慎改动！
 */
package socket

import (
	"github.com/imroc/req/v3"
)
import chat "github.com/sheason2019/linkme/omi/chat"

type ChatSocketClient struct {
	Request *req.Client
	HOST    string
}

func (ChatSocketClient) New(host string) (definition ChatSocketClient) {
	definition.HOST = host
	return
}
func (definition ChatSocketClient) GetRequestClient() *req.Client {
	if definition.Request != nil {
		return definition.Request
	}
	return req.C()
}
func (definition ChatSocketClient) PostUserSequence(userSequence []UserConversationSequence) {
	client := definition.GetRequestClient()
	resp, err := client.R().SetBody(&PostUserSequenceRequest{UserSequence: userSequence}).Post(definition.HOST + "/ChatSocket.UserSequence")
	if err != nil {
		panic(err)
	}
	if resp.IsError() {
		panic("远程调用错误")
	}
	return
}

func (definition ChatSocketClient) PostMessages(convId int, messages []chat.Message) {
	client := definition.GetRequestClient()
	resp, err := client.R().SetBody(&PostMessagesRequest{ConvId: convId, Messages: messages}).SetBody(&PostMessagesRequest{ConvId: convId, Messages: messages}).Post(definition.HOST + "/ChatSocket.Messages")
	if err != nil {
		panic(err)
	}
	if resp.IsError() {
		panic("远程调用错误")
	}
	return
}

func (definition ChatSocketClient) KickoutMember(members []chat.MessageMember) {
	client := definition.GetRequestClient()
	resp, err := client.R().SetBody(&KickoutMemberRequest{Members: members}).Post(definition.HOST + "/ChatSocket.KickoutMember")
	if err != nil {
		panic(err)
	}
	if resp.IsError() {
		panic("远程调用错误")
	}
	return
}

func (definition ChatSocketClient) ConversationUpdate(convId int) {
	client := definition.GetRequestClient()
	resp, err := client.R().SetBody(&ConversationUpdateRequest{ConvId: convId}).Post(definition.HOST + "/ChatSocket.ConversationUpdate")
	if err != nil {
		panic(err)
	}
	if resp.IsError() {
		panic("远程调用错误")
	}
	return
}

File: web/src/api-lib/socket-client.ts
/**
 * 本文件由Omi.js自动生成，谨慎改动！
 * 生成时间：2022年11月28日 22:56:17.
 */

import { OmiClientBase } from "@omi-stack/omi-client";
import { AxiosRequestConfig } from "axios";

import { SequenceItem, Message, MessageMember } from "./chat-client";
export interface UserConversationSequence {
  UserId: number;
  Sequence: SequenceItem[];
}
export class ChatSocketClient extends OmiClientBase {
  PostUserSequence(userSequence: UserConversationSequence[]) {
    const url = "ChatSocket.UserSequence";
    const method = "Post";
    return this.request<void>(url, method, { userSequence });
  }
  PostMessages(convId: number, messages: Message[]) {
    const url = "ChatSocket.Messages";
    const method = "Post";
    return this.request<void>(url, method, { convId, messages });
  }
  // 删除成员
  KickoutMember(members: MessageMember[]) {
    const url = "ChatSocket.KickoutMember";
    const method = "Post";
    return this.request<void>(url, method, { members });
  }
  // 会话信息发生了变化，让正在会话中的成员重新同步会话信息
  ConversationUpdate(convId: number) {
    const url = "ChatSocket.ConversationUpdate";
    const method = "Post";
    return this.request<void>(url, method, { convId });
  }
}
